### PR TITLE
update_dq improvement in findsat_mrt

### PR DESCRIPTION
**Pull request by dvstark**
### Description

This PR is to add in functionality to acstools.utils_findsat_mrt.updatedq to expand the input mask to match the size of the original DQ array if necessary. This is needed if a satellite trail was identified and masked on a binned image, which is common. Users could do this manually, but it is needed frequently enough that it made sense to include. This step has also been designed to handle images with odd dimensions,(e.g., WFC3/UVIS are 2051x4096) which ensures the code is applicable to other imaging data. 

Warnings have been added into acstools.findsat_mrt.TrailFinder.rebin step to inform users if their data has odd dimensions and that rows/columns were being trimmed as needed to allow rebinning (this was happening already, but users may not have been aware). These trimmed rows/columns are added back in during the update_dq step after the satellite mask is enlarged.

### CODE REVIEW

**Comment by pllim** [comment]
Lemme close/reopen and see.

**Comment by pllim** [comment]
You can ignore the triage failure in this PR.  That is my bad.

**Comment by dvstark** [comment]
just to put all comments in one space: I had asked what the triage step is and what might be causing it to fail

File: acstools/findsat_mrt.py
Comment (pllim) [inline acstools/findsat_mrt.py:1365]:
```suggestion
        if (((self.image.shape[0] % self.binsize != 0)) |
               ((self.image.shape[1] % self.binsize != 0))):
```

File: acstools/findsat_mrt.py
Comment (pllim) [inline acstools/findsat_mrt.py:1367]: No need to put "Warning:" because logger will make that clear.
```suggestion
            LOG.warning('Image dimensions do not evenly divide by bin '
                        'size.\nExtra rows/columns will be trimmed first.')
```

File: acstools/utils_findsat_mrt.py
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1203]:
```suggestion
    expand_mask : bool, optional
```

File: acstools/utils_findsat_mrt.py
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1206]:
```suggestion
        the original image.
```

File: acstools/utils_findsat_mrt.py
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1223]:
```suggestion
                            'original size.')
```

File: acstools/utils_findsat_mrt.py
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1226]:
```suggestion
                LOG.info('Enlarging mask to original size.')
```

File: acstools/utils_findsat_mrt.py
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1227]: The logger saying "enlarging" but I don't see any checks on whether one dimension is really larger than the other one. Won't it be shrinking in some cases? Should that be allowed?

File: acstools/utils_findsat_mrt.py
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1236]: Instead of doing it twice, is there a smarter way to check for odd/even at the start and then go different logic routes from there?

**Review by pllim — COMMENTED** [review]
You do have a minimal test in https://github.com/spacetelescope/acstools/blob/master/acstools/tests/test_wfc_findsat_mrt.py . If you can expand your test suite to cover this new option, would prevent regression in the future, but I'll leave that up to you.

File: acstools/utils_findsat_mrt.py
Comment (jryon) [reply acstools/utils_findsat_mrt.py:1227]: Seconding this. I did a couple of tests and the error for a mask that is bigger than the DQ array is pretty cryptic. If you never expect this case to actually happen, should handle it more elegantly.

**Review by jryon — CHANGES_REQUESTED** [review]
I ran a few basic tests with different sizes of masks paired with ACS and WFC3 data. The results look good and make sense, except for the case when the mask is larger than the DQ array.

File: acstools/utils_findsat_mrt.py
Comment (dvstark) [reply acstools/utils_findsat_mrt.py:1227]: Thanks y'all. I can't think of a situation where someone would want to supersample the original image to make it larger for satellite trail detection (feel free to chime in if you think I'm thinking too narrowly). I think the best path forward is to enforce self.binsize > 1, which will always yield a mask smaller or equal in size to the DQ array. I can add some checks inside the updatedq() function to check this as well.

File: acstools/utils_findsat_mrt.py
Comment (dvstark) [reply acstools/utils_findsat_mrt.py:1236]: I cleaned this up a bit, but it's more like an extra step if the images don't divide evenly, rather than completely different logic

**Comment by dvstark** [comment]
I'm unsure what to do about the error in the "Python 3.12 with remote data and dev dependencies" CI test:py312-test-

```
devdeps: commands[3] /home/runner/work/acstools/acstools/.tmp/py312-test-devdeps> pytest --pyargs acstools /home/runner/work/acstools/acstools/doc --remote-data -v
ImportError while loading conftest '/home/runner/work/acstools/acstools/conftest.py'.
../../conftest.py:10: in <module>
    from acstools.version import version
../../acstools/__init__.py:20: in <module>
    from . import satdet  # noqa
../../acstools/satdet.py:152: in <module>
    import skimage
../../.tox/py312-test-devdeps/lib/python3.12/site-packages/skimage/__init__.py:128: in <module>
    from ._shared import geometry
geometry.pyx:1: in init skimage._shared.geometry
    ???
E   ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
py312-test-devdeps: exit 4 (1.01 seconds) /home/runner/work/acstools/acstools/.tmp/py312-test-devdeps> pytest --pyargs acstools /home/runner/work/acstools/acstools/doc --remote-data -v pid=2518
  py312-test-devdeps: FAIL code 4 (303.45=setup[272.30]+cmd[9.07,20.65,0.42,1.01] seconds)
  evaluation failed :( (303.59 seconds)
Error: Process completed with exit code 4.
```

Some new issue with skimage and satdet?

**Comment by pllim** [comment]
You can ignore the devdeps failure for now. Maybe scikit-image nightly wheel needs time to catch up to numpy nightly wheel. With numpy 2.0 in beta now, hopefully soon. I'll give it a few days and up to a week.

**Comment by dvstark** [comment]
I've done the updates and am ready for this PR to be reviewed again. Thanks for the feedback so far!

File: acstools/utils_findsat_mrt.py
Comment (jryon) [inline acstools/utils_findsat_mrt.py:1226]: need a space between 'rescale' and 'masks'

File: acstools/utils_findsat_mrt.py
Comment (jryon) [inline acstools/utils_findsat_mrt.py:1251]: need spaces between 'by' and 'different, 'the' and 'DQ'

Comment (jryon) [reply acstools/utils_findsat_mrt.py:1251]: I tried a square mask (1000x1000), and got this error. I think the error could be more helpful, but I'm not sure how. Perhaps something to do with the aspect ratio of the mask must be close to that of the DQ array?

**Review by jryon — COMMENTED** [review]
Couple more very minor comments

File: acstools/utils_findsat_mrt.py
Comment (dvstark) [reply acstools/utils_findsat_mrt.py:1251]: thanks Jenna. Do you meant the error message itself could be more helpful? It could say something like "Mask aspect ratio too different from DQ array" for instance.

Comment (jryon) [reply acstools/utils_findsat_mrt.py:1251]: Yes, that's what I mean. Essentially, the message should help the user figure out how to fix the problem. Could think about printing out the aspect ratio of the mask and the dq array, and tell the user to make them more similar?

Comment (dvstark) [reply acstools/utils_findsat_mrt.py]: Ok, I modified that error message. Let me know what you think

Comment (jryon) [reply acstools/utils_findsat_mrt.py]: Looks good to me!

**Review by jryon — APPROVED** [review]
Happy with this!

**Comment by dvstark** [comment]
@pllim The Wheel-building step is being skipped. Do we need to make sure that passes before merging this in?

**Comment by pllim** [comment]
Does not hurt. I added the label to enable it.

File: acstools/utils_findsat_mrt.py
Comment (pllim) [inline acstools/utils_findsat_mrt.py:1203]:
```suggestion
    expand_mask : bool, optional
```

**Review by pllim — COMMENTED** [review]
Since Jenna approved, I have nothing much to add, except one Sphinx formatting fix.

**Pull request merged**
